### ISSUE TITLE
Double nested PropertyTypes are not checked correctly

### ISSUE THREAD

**Issue by martysweet** [martysweet/cfn-lint#15]
A template such as 

```yaml
Resources:
  CloudFrontDistribution:
    Type: "AWS::CloudFront::Distribution"
    Properties:
      DistributionConfig:
        Aliases:
          - !Ref DomainName
        CacheBehaviors:
          CacheBehavior
        DefaultCacheBehavior:
          DefaultCacheBehavior
        DefaultRootObject: index.php"
        Enabled: true
        Logging:
          Bucket: mybucket
          IncludeCookies: false
          Prefix: !Sub "${DomainName}/"
       Origins:
          - Id: !Sub "Root-${OriginDomainName}"
            DomainName: !Ref OriginDomainName
            CustomOriginConfig:
              OriginProtocolPolicy: http-only
        PriceClass: String
        ViewerCertificate:
          AcmCertificateArn: !Ref CertificateArn
```

Will throw an error of:
```
Resource: Resources > CloudFrontDistribution > Properties > DistributionConfig
Message: Bucket is not a valid property of AWS::CloudFront::Distribution.DistributionConfig.Logging
Documentation: http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-cloudfront-distributionconfig.html#cfn-cloudfront-distributionconfig-logging

Resource: Resources > CloudFrontDistribution > Properties > DistributionConfig
Message: CustomOriginConfig is not a valid property of AWS::CloudFront::Distribution.DistributionConfig.Origin
Documentation: http://docs.aws.amazon.com/search/doc-search.html?searchPath=documentation-guide&searchQuery=AWS::CloudFront::Distribution.DistributionConfig.Origin&x=0&y=0&this_doc_product=AWS+CloudFormation&this_doc_guide=User+Guide&doc_locale=en_us#facet_doc_product=AWS%20CloudFormation&facet_doc_guide=User%20Guide
```

Among others...

This should be `Resources > CloudFrontDistribution > Properties > DistributionConfig > Logging`, however it looks like the PropertyType is correct.
